clustermesh-apiserver/clustermesh/converters.go

```diff
@@ -218,12 +257,20 @@ func newCiliumEndpointSliceOptions(cfg cmk8s.CiliumEndpointSliceConfig) Options[
 	}
 }
 
-func newCiliumEndpointSliceConverter(logger *slog.Logger, cinfo cmtypes.ClusterInfo) Converter[*cilium_api_v2a1.CiliumEndpointSlice] {
-	return NewCachedCoverter(ciliumEndpointSliceMapper)
+func newCiliumEndpointSliceConverter(logger *slog.Logger, cinfo cmtypes.ClusterInfo, tracker GlobalNamespaceTracker) Converter[*cilium_api_v2a1.CiliumEndpointSlice] {
+	nsFilter := NewGlobalNamespaceFilter(tracker)
+	return NewCachedCoverter(func(endpointslice *cilium_api_v2a1.CiliumEndpointSlice) iter.Seq[store.Key] {
+		return ciliumEndpointSliceMapper(endpointslice, nsFilter)
+	})
 }
 
-func ciliumEndpointSliceMapper(endpointslice *cilium_api_v2a1.CiliumEndpointSlice) iter.Seq[store.Key] {
+func ciliumEndpointSliceMapper(endpointslice *cilium_api_v2a1.CiliumEndpointSlice, nsFilter NamespaceFilter) iter.Seq[store.Key] {
 	return func(yield func(store.Key) bool) {
+		// Only export endpoint slices from global namespaces
```

Comment:
nit so that it doesn't get confused with regular k8s epslice
```suggestion
		// Only cilium export endpoint slices from global namespaces
```